cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/SynchronousMachineConversion.java

```diff
@@ -55,6 +56,13 @@ public void convert() {
         Generator g = adder.add();
         convertedTerminals(g.getTerminal());
         convertReactiveLimits(g);
+        if (p.asInt("referencePriority", 0) > 0) {
+            if (g.getTerminal().getBusBreakerView().getBus() == null) {
+                context.ignored("referencePriority", () -> String.format("%s is disconnected from the network", g.getId()));
+            } else {
+                SlackTerminal.attach(g.getTerminal().getBusBreakerView().getBus());
```

Comment:
@MioRtia can you here not use the ````attach```` method but the ````set```` one in provinding directly the connected terminal of the generator ? With that, you can always set the extension, no ?

Reply:
@zamarrenolm This will solve your third point (the terminal defined will be the terminal of the equipment). For the other point, since I think the case where a voltage level contains two slack terminals of different synchronous components is rare, couldn't we ignore this case? Maybe we should look at how often it happens in real life.